atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ProfilingKeyValueService.java

```diff
@@ -84,6 +86,44 @@ public static ProfilingKeyValueService create(KeyValueService delegate, long slo
         void log(String fmt, Object... args);
     }
 
+    interface FlushableLoggingFunction extends LoggingFunction, AutoCloseable {
+        void flush();
+
+        @Override
+        void close();
+    }
+
+    // Accumulates logs in a single string.
+    // Warning to users of this class: We do not guarantee that SLF4J special characters work properly across log lines,
+    // nor do we guarantee behaviour when the number of arguments does not match the number of placeholders.
+    @VisibleForTesting
+    static class LogAccumulator implements FlushableLoggingFunction {
+        private static final String DELIMITER = "\n";
+
+        private final List<String> formatElements = Lists.newArrayList();
+        private final List<Object> argList = Lists.newArrayList();
+        private final LoggingFunction sink;
+
+        LogAccumulator(LoggingFunction sink) {
+            this.sink = sink;
+        }
+
+        @Override
+        public synchronized void flush() {
```

Comment:
Shouldn't flush actually truncate the accumulators?

Reply:
yeah, we should probably just get rid of `flush()` and just write logs on `close()` only